rfcs/0013-synthea-future-claims-and-automation.md

```diff
@@ -0,0 +1,139 @@
+
+  - Future Claim Data - For both quarterly and weekly, a batch of similar size for claim data with a future date will also be generated in a similar fashion. 
+
+  - Special data parameters of interest: One parameter, which has not changed in previous releases of Synthea is Part D Event ID (PDE ID). Despite monotonically increasing other parameters i.e. bene ID, claim group ID, the value of PDE ID will not need to change on a regular basis, unless there is a specific customer use case presented
+
+Automated Recurring Loading: 
+  - Once the synthetic data is generated, within the same hosted cloud instance, the latest version of the BFD application from github, and dockerized database, will have been installed prior, and will be used to run the BFD pipeline integration test (IT), to load the new batch of synthetic data, ensuring that the newly generated data can be inserted properly.
+
+  - Additionally, the RIF files from the previous weekly/quarterly batch of Synthea data will be pulled from AWS S3, and loaded into the database with the new batch of synthea data.  
+
+  - Once the automated load testing on the instance is complete, a script will run that modifies the new batch's generated manifest file, and uploads the RIF files, end-state properties, and manifest files to S3 via command line file transfer. This will trigger the BFD ingestion pipeline in PROD SBX or TEST so that the data is stored in the database.
+
+  - The separate RIF files associated with future claim data for the given release will be put uploaded in a specially designated folder in S3 for staging until loading in the future. 
+
+  - Additional testing - As of 04/21/2022, there is a plan in place to remve tens of thousands of old synthetic beneficiary data prior to Synthea data being stored in the database, which is largely responsible for the database collisions that have taken place in the past, when loading new Synthea data. Once this data is permenantly removed, it will likely take many years for there to be enough data to cause overlap. So after making sure the data generated does not collide with the previous batch of Synthea data, it should be safe to load.  
+
+Large Synthetic Data Generation:
+  - Each PI (Program Increment), approximately every 8 weeks, 60 million Synthea beneficiaries will be generated and available for on-demand performance and load testing on TEST and PROD SBX databases. The number of beneficiaries in the dataset size is meant to reflect the size and shape of production. The process will be similar to that of automated recurring generation, however the memory and computational power of the AWS instance will be larger for time and cost effectiveness. 
+  
```

Comment:
Just a thought; if we continually increase the DB size over time, performance will slowly degrade just from the slowness of having more data to trawl through on each search (compare prods slow query times vs the lower environments near-instant searches due to data volume difference.) It's something we'll need to keep in mind if we expect to continually dump 60 million new benes every few months.